src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public abstract class Figure implements FiguresCalculations, FiguresDrawers {
+    protected String color;
+
+    public Figure() {
+        this.color = new ColorSupplier().getRandomColor();
```

Comment:
Consider moving the random color generation to the `FigureSupplier` class, as per the checklist recommendation: 'You can pass random values to the constructor of a figure instead of generating them inside figure classes.' This will make your code more modular and adhere to the task requirements.